File: lab1/lab1.py
from typing import Any
from numbers import Number


def isEmpty(L: list) -> bool:
    return len(L) == 0


def head(L: list) -> Any:
    return L[0]


def tail(L: list) -> list:
    return L[1:]


def sum(L: list) -> Number:
    if(isEmpty(L)):
        return 0
    else:
        return head(L) + sum(tail(L))


arr = [1, 2, 3, 4]
print(sum(arr))

File: lab1/lab1.js
function isEmpty(array)
{
    return array.length()==0;
}

function head(array)
{
    return array[0];
}

function tail(array)
{
    return array.slice(1)
}

ar= [1,2,3,4]

sum(ar)

function sum(array)
{
    if(isEmpty(array)) return 0;
    return head(array) + sum(tail(array))
}

function sumT(array,sum)
{
    if(isEmpty(array)) return sum

}
function sum2(array)
{
    sum(T(array,0));
}